test/models/driver_test.rb

```diff
@@ -50,32 +51,68 @@
     it "must have a VIN number" do
       # Arrange
       new_driver.vin = nil
+      new_driver.save
 
       # Assert
       expect(new_driver.valid?).must_equal false
       expect(new_driver.errors.messages).must_include :vin
       expect(new_driver.errors.messages[:vin]).must_equal ["can't be blank"]
     end
+
+    it "must have a unique VIN number" do
+      # Arrange
+      new_driver.save
+      conflicting_driver = Driver.new(name: "Cathy", vin: "123", available: true)
+
+      # Assert
+      expect(conflicting_driver.valid?).must_equal false
+      expect(conflicting_driver.errors.messages).must_include :vin
+      expect(conflicting_driver.errors.messages[:vin]).must_equal ["has already been taken"]
+    end
+
   end
 
   # Tests for methods you create should go here
   describe "custom methods" do
     describe "average rating" do
-      # Your code here
+      it "calculates the average rating" do
+        # Arrange
+        new_driver.save
+        new_passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+        new_passenger.save
+        # Assert - No trips
+        expect(new_driver.avg_rating).must_be_nil
+        # Act and Assert - 1 trip
+        trip_1 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 5)
+        trip_1.save
+        new_driver.reload
+        expect(new_driver.avg_rating).must_equal 5
+        # Act and Assert - 2 trips
+        trip_2 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, rating: 3)
+        new_driver.reload
+        expect(new_driver.avg_rating).must_equal 4
+      end
     end
-
     describe "total earnings" do
-      # Your code here
-    end
-
-    describe "can go online" do
-      # Your code here
-    end
-
-    describe "can go offline" do
-      # Your code here
+      it "calculates the total earnings" do
+        # Arrange
+        new_driver.save
+        new_passenger = Passenger.create(name: "Kari", phone_num: "111-111-1211")
+        new_passenger.save
+        # Assert - No trips
+        expect(new_driver.total_earnings).must_equal 0
+        # Act and Assert - 1 trip
+        trip_1 = Trip.create(driver_id: new_driver.id, passenger_id: new_passenger.id, date: Date.today, cost: 200)
+        trip_1.save
```

Comment:
This is just a small nitpick:

`Trip.new` creates an instance of `Trip`
`Trip.create` creates an instance of trip and then calls `.save` on that instance.

Therefore, `Trip.create(...)` and then `trip_1.save` is redundant! I would either just use one line with `.create` or use two lines with `.new` then `.save`